corehq/form_processor/management/commands/set_meta_fields.py

```diff
@@ -0,0 +1,103 @@
+import logging
+import datetime
+
+from bulk_update.helper import bulk_update as bulk_update_helper
+from django.core.management.base import BaseCommand
+from django.db.models import Q
+
+from corehq.form_processor.backends.sql.dbaccessors import FormAccessorSQL
+from corehq.form_processor.models import XFormInstanceSQL
+from corehq.sql_db.util import split_list_by_db_partition
+from dimagi.utils.chunked import chunked
+
+
+logger = logging.getLogger('set_meta_fields')
+logger.setLevel('DEBUG')
+
+META_FIELDS = ("time_end", "time_start", "commcare_version", "build_version")
+
+
+def sql_iter_forms(form_ids):
+    for chunk in chunked(form_ids, 100):
+        chunk = list([_f for _f in chunk if _f])
+        for form in FormAccessorSQL.get_forms(chunk):
+            yield form
+
+
+def form_ids_by_received_on(from_date, to_date):
+    from corehq.sql_db.util import run_query_across_partitioned_databases
+    q_expr = Q(received_on__gte=from_date, received_on__lte=to_date)
+    needs_update_q = (
+        Q(time_end__isnull=True) &
+        Q(time_start__isnull=True) &
+        Q(commcare_version__isnull=True) &
+        Q(build_version__isnull=True)
+    )
+    q_expr = q_expr & needs_update_q
+    return list(run_query_across_partitioned_databases(
+        XFormInstanceSQL, q_expr, values=['form_id']
+    ))
+
+
+def set_meta_fields(from_date, to_date, failfast):
+    all_form_ids = form_ids_by_received_on(from_date, to_date)
+    if not all_form_ids:
+        print "No remaining forms to be migrated!"
+        return
+    total_count = len(all_form_ids)
+    updated_count = 0
+    # split by partition for bulk updating
+    for dbname, form_ids_by_db in split_list_by_db_partition(all_form_ids):
+        for form_ids in chunked(form_ids_by_db, 100):
+            forms_to_update = []
+            for form in sql_iter_forms(form_ids):
```

Comment:
seems like you're already chunking the form_ids so it shouldn't be necessary to do it again in `sql_iter_forms`

Reply:
ha, thanks, I will remove the line 51